### PR TITLE
php: Add devel version for PHP 7.3

### PR DESCRIPTION
**Pull request by kabel**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Provides a baseline for upgrading to a new minor version and allows devs to test for migration.

### CODE REVIEW

**Comment by kabel** [comment]
Build is failing due to new ldap symbols being used in 7.3 that are not available in the macOS ldap framework. Could be worked around by depending on `openldap` which does have these symbols, but then we'd have the same problem with the linker in #28398.

I can report upstream, but I have doubts about their support of macOS's stripped down libldap API.

**Comment by SMillerDev** [comment]
Reporting it can't hurt I guess.

**Comment by kabel** [comment]
Upstream report at https://bugs.php.net/bug.php?id=76433

**Comment by kabel** [comment]
I'd say so. I still can't figure out how/why the symbols are in the header file but abscent in the lib.

**Comment by javian** [comment]
It feels like we need to get the issue with the incorrectly linked libraries fixed so we can work around this and the other issues we've had since the openldap support in macos is unlikely to get corrected. I'm trying to find a solution to showing the issue outside of the homebrew formula when doing a manual compilation from source.

**Comment by javian** [comment]
@kabel you should be able to commit this now without making libxml changes since we depend on the homebrew library now.

File: Formula/php.rb [hunk 1]
```diff
@@ -29,6 +34,7 @@ class Php < Formula
   depends_on "libpq"
   depends_on "libsodium"
   depends_on "libzip"
+  depends_on "openldap"
```
Comment (javian) [inline Formula/php.rb]: Make this a devel dependency.

File: Formula/php.rb [hunk 2]
```diff
@@ -161,7 +173,7 @@ def install
     extension_dir = Utils.popen_read("#{bin}/php-config --extension-dir").chomp
     orig_ext_dir = File.basename(extension_dir)
     inreplace bin/"php-config", lib/"php", prefix/"pecl"
-    inreplace "php.ini-development", "; extension_dir = \"./\"",
+    inreplace "php.ini-development", %r{; ?extension_dir = "\.\/"},
```
Comment (javian) [inline Formula/php.rb]: As far as I can tell you don't have to escape the `/` when using %r

**Comment by javian** [comment]
@kabel yes I don't think we can accomodate that until we create the new formula. Assuming php works as expected then the users would have to do their own lifting. I can't see any other breaking changes after a quick glance through the upgrade documentation.

**Comment by zbeekman** [comment]
Is there a good reason not to combine this with PR #30446? Or can we merge #30446 while we wait on changes here?

I'm a bit out of my depth when it comes to PHP, so I'll let @javian or someone else weigh in.

File: Formula/php.rb [hunk 3]
```diff
@@ -153,6 +159,12 @@ def install
       args << "--with-curl"
     end
 
+    if devel?
```
Comment (DomT4) [inline Formula/php.rb]: `build.devel?` rather than `devel?` everywhere outside of `test do` blocks, please.

**Review by javian — APPROVED** [review]
Lets ship it, did you have any final remarks @DomT4 ?

**Comment by DomT4** [comment]
Needs squashing on merge but that can be done when you merge/pull/whatever method of merging this into `master` you choose 😄. LGTM, ship away 👍.

**Comment by javian** [comment]
Thanks @kabel ❤️ keep em coming ^^

**Pull request closed without merging**